src/libstd/collections/hash/table.rs

```diff
@@ -581,6 +599,16 @@ impl<'t, K, V> FullBucket<K, V, &'t mut RawTable<K, V>> {
             v)
         }
     }
+
+    /// Remove this bucket's key and value from the hashtable.
+    /// Only used for inplacement insertion.
+    pub fn remove(&mut self) {
+        self.table.size -= 1;
+        unsafe {
+            *self.raw.hash = EMPTY_BUCKET;
+            ptr::read(self.raw.pair); // drop right now
```

Comment:
Note: that the code might not reproduce exactly the way I described it, but it is still reading uninitialized memory.